### ISSUE TITLE
Incorrect route line rendering

### ISSUE THREAD

**Issue by PaulStets** [osmandapp/Osmand#4294]
The issue occurs with all OsmAnd versions on this device:
![screenshot_2017-08-11-09-51-39-231_com android settings](https://user-images.githubusercontent.com/20681525/29208942-c4cfd6f2-7e94-11e7-9f3c-c80294eafa4a.png)

Screenshots:
![screenshot_2017-08-10-09-54-48-488_net osmand plus 1](https://user-images.githubusercontent.com/20681525/29208961-d17369be-7e94-11e7-81cd-c81114f381fb.png)
![screenshot_2017-08-10-09-55-03-797_net osmand plus](https://user-images.githubusercontent.com/20681525/29208962-d17653ea-7e94-11e7-86d4-2c55738b59ab.png)

**Comment by sonora** [comment]
Cross reference: I guess these are all the same: #2024, #2035, #2157.

We suspected device specific video driver issues at some point.

**Comment by sonora** [comment]
Sometimes these kinds of issues may be circumvented by disabling certain hardware acceleration features (possibly also in our code, not sure). The main things to try on the device should be if this can be worked around by disabling power saving features, and in the device developer settings toggling things like "Disable HW overlays", or perhaps "Force GPU rendering".

Searching the web for "Android screen distortion" will lead to more occurrences and clues, some of which look related.

**Comment by landswellsong** [comment]
Hi,
Been suffering with it for long too. The phone is Xiaomi Redmi Note 2. Still reproducible on `master` built. OpenGL options in debugging don't seem to have any effect neither has switching native off.

That being said, I noticed the same effect with the regional outlines which led me to believe it has something to do with polygons having non-solid alpha fill (otherwise normal roads would be affected). This is further confirmed by how it affects the night mode, as you can see only the outline is affected (day mode `route` doesn't have a distinguishable outline as per default rendering setting):
![nightmode](https://user-images.githubusercontent.com/2434329/37724233-f8f380b6-2d38-11e8-8b9d-652f541e5f98.png)

I'd love to fix it definitely at some point, but my schedule doesn't permit me to at the moment. So I did a quick rendering style fix and it did have an effect. Here, same scene, left is default rendering style, right is my fix:
![before_after](https://user-images.githubusercontent.com/2434329/37724040-9613064c-2d38-11e8-97c5-c401d0c0d1c8.png)

Before it's conclusively fixed I guess some users might benefit from a temporary remedy. The file I used:
```XML
<renderingStyle name="Depends-template" depends="default" defaultColor="#f1eae4" version="1">
	<!-- In a dependent renderer, just specify what should change vs. the master/default renderer -->

	<renderingProperty attr="appMode" name="Rendering mode" description="Map optimization for respective User Profile"
		type="string" possibleValues="browse map,car,bicycle,pedestrian"/>

	<renderingConstant name="null" value="#00000000"/>

    <renderingAttribute name="route">
        <case color="#2a4bd1" strokeWidth="12:8" color_3="#ffde5b" color_2="#ff0000" strokeWidth_3="5:7">
            <apply_if nightMode="true" color="#ffdf3d" color_2="#00ff00" color_3="#41a6d9" strokeWidth="9:8" color_0="#b29c2b" strokeWidth_0="12:8"/>
        </case>
    </renderingAttribute>
</renderingStyle>
```
Basically copied over from `default` with alpha values removed. 

Note this only fixes the `route`, regional outlines are still broken, I was lazy to look up where their color is being set and making them solid would mess up the map even more than having the distorted.

Hope that helps. When I get some free time I can try debugging it myself, but not sure it will be in near future.

PS: sorry posted in the wrong thread by mistake

**Comment by landswellsong** [comment]
@vshcherb yes, exactly. Non-solid alpha causes glitches. The green "downloaded region" fill on low zoom levels exhibits same artifacts as the roads in OP. 

I'll try to set the same on the normal roads and see if it has any effect.

**Comment by landswellsong** [comment]
Roads and water seems unaffected, set a nonsolid alpha to them and here's the results
![screenshot_net osmand plus_2018-03-27-22-53-10](https://user-images.githubusercontent.com/2434329/37994383-ab90097e-3211-11e8-8dfd-9447c4ed2079.png)

![screenshot_net osmand plus_2018-03-27-22-49-25](https://user-images.githubusercontent.com/2434329/37994297-6afaf5fe-3211-11e8-91b1-a836e094fc6f.png)
